### PR TITLE
Add SubscriptionRequests trait.

### PR DESCRIPTION
**Pull request by Ruveloc02**
### **What does this PR do?** :robot:
:new: Add SubscriptionRequests trait to subscription package in order to make requests directly from a Subscription model object.
:new: Add ConfiguresGateway trait to subscription package.

### CODE REVIEW

File: src/Models/Traits/ConfiguresSubscriptionGateway.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    /**
+     * Retrieve the subscription's configured gateway.
+     *
+     * @return \Payavel\Subscription\SubscriptionGateway
+     */
+    public function getGatewayAttribute()
+    {
+        if (! isset($this->subscriptionGateway)) {
+            $this->subscriptionGateway = (new SubscriptionGateway);
```
Comment (r-kujawa) [inline src/Models/Traits/ConfiguresSubscriptionGateway.php:24]: I will push the fix to be able to retrieve the provider with the config driver, and let you know when that is complete so you can update.

Comment (r-kujawa) [reply src/Models/Traits/ConfiguresSubscriptionGateway.php:24]: Completed in https://github.com/payavel/subscription/pull/21. You may proceed on configuring the gateway based on `provider_id`.

Comment (Ruveloc02) [reply src/Models/Traits/ConfiguresSubscriptionGateway.php]: Done!

**Review by r-kujawa — APPROVED** [review]
🚀 LGTM!

**Pull request merged**